ci/cloudbuild/builds/cmake-gcs-rest-install.sh

```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
+#
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+source "$(dirname "$0")/../../lib/init.sh"
+source module ci/cloudbuild/builds/lib/cmake.sh
+source module ci/cloudbuild/builds/lib/quickstart.sh
+source module ci/cloudbuild/builds/lib/features.sh
+source module ci/lib/io.sh
+
+export CC=clang
+export CXX=clang++
+
+INSTALL_PREFIX="$(mktemp -d)"
+readonly INSTALL_PREFIX
+
+read -r ENABLED_FEATURES < <(features::list_full_cmake)
+mapfile -t cmake_args < <(cmake::common_args)
+
+# Compiles and installs all libraries and headers.
+cmake "${cmake_args[@]}" \
+  -DCMAKE_INSTALL_PREFIX="${INSTALL_PREFIX}" \
+  -DCMAKE_INSTALL_MESSAGE=NEVER \
+  -DBUILD_TESTING=OFF \
+  -DGOOGLE_CLOUD_CPP_ENABLE_EXAMPLES=OFF \
+  -DGOOGLE_CLOUD_CPP_ENABLE="${ENABLED_FEATURES}"
+cmake --build cmake-out
+cmake --install cmake-out --component google_cloud_cpp_development
+
+io::log_h2 "Verifying installed directories"
+# Finds all the installed leaf directories (i.e., directories with exactly two
+# links: . and ..). We only look at leaf directories, because obviously all
+# parent directories must also exist.
+mapfile -t actual_dirs < <(env -C "${INSTALL_PREFIX}" find -type d)
+# The micro generator populates `ci/etc/expected_install_directories` with the
+# expected directories generated via `make install`. This generated list does
+# not include:
+#   - hand-crafted libraries
+#   - directories with pure protos (without gRPC code)
+# We maintain those here as a hard-coded list.
+mapfile -t expected_dirs < <(cat ci/etc/expected_install_directories)
+expected_dirs+=(
```

Comment:
I assume this list must match the list in `cmake-install.sh`?  Can we refactor it to a text file and load it from both scripts?